aiida_mlip/data/model.py

```diff
@@ -0,0 +1,159 @@
+"""Define Model Data type in AiiDA"""
+
+import hashlib
+from pathlib import Path
+from urllib import request
+from urllib.parse import urlparse
+
+from aiida.orm import SinglefileData
+
+
+def _calculate_hash(file: str | Path) -> str:
```

Comment:
`|` is incompatible with Python 3.9, unless you also include `from __future__ import annotations`. (We decided against using that in janus-core for now, as it seemed to lead to issues with ASE, but there may be other solutions.)

An alternative is `Union`.

The CI should have picked up on this, but you can also test locally with `tox run -e py39` for Python 3.9 tests, or `tox run -e ALL` for all tests.